mvnw

```diff
@@ -282,35 +251,58 @@ fi
 # End of extension
 ##########################################################################################
 
-export MAVEN_PROJECTBASEDIR=${MAVEN_BASEDIR:-"$BASE_DIR"}
-if [ "$MVNW_VERBOSE" = true ]; then
-  echo $MAVEN_PROJECTBASEDIR
+# If specified, validate the SHA-256 sum of the Maven wrapper jar file
+wrapperSha256Sum=""
+while IFS="=" read -r key value; do
+  case "$key" in (wrapperSha256Sum) wrapperSha256Sum=$value; break ;;
+  esac
+done < "$MAVEN_PROJECTBASEDIR/.mvn/wrapper/maven-wrapper.properties"
+if [ -n "$wrapperSha256Sum" ]; then
+  wrapperSha256Result=false
+  if command -v sha256sum > /dev/null; then
+    if echo "$wrapperSha256Sum  $wrapperJarPath" | sha256sum -c > /dev/null 2>&1; then
+      wrapperSha256Result=true
+    fi
+  elif command -v shasum > /dev/null; then
```

Comment:
Afaiu, the checkcum validation works on both Linux and MacOS. Does it make sense to enable `wrapperSha256Sum` in the `maven-wrapper.properties` by default. Windows user will get error message. (NIT: maybe a specific error message for windows users and remove it after FLINK-33607 has been fixed.) WDYT?

Reply:
That's auto-generated code (`./mvnw wrapper:wrapper` was called). I'm hesitant to add customizations to auto-generated code and would rather prefer to fix it upstream.

Reply:
fair enough, thanks for the info.